### PR TITLE
Refactor GZIP encoding check

### PR DESCRIPTION
**Pull request by ndegwamartin**
## Description of what I changed

When determining whether to apply the Gzip header [our condition](https://github.com/google/fhir-gateway/blob/4376a58fa2e1316050177c3bb3d206f1f1fcd7c1/server/src/main/java/com/google/fhir/gateway/BearerAuthorizationInterceptor.java#L214) checks for an exact match in the header value passed for **Accept-Encoding**. This is limiting since various clients by default supply multiple comma separated values _e.g. Accept-Encoding: gzip, deflate, br_ and the server responds according to configuration. 

A better approach would be to use the `String.contains` method to check for the occurrence of the GZip literal.

## E2E test

TESTED:

Please replace this with a description of how you tested your PR beyond the
automated e2e/unit tests.

## Checklist: I completed these to help reviewers :)

- [x] I have read and will follow the [review process](https://github.com/GoogleCloudPlatform/openmrs-fhir-analytics/blob/master/doc/review_process.md).
- [x] I am familiar with Google Style Guides for the language I have coded in.

  No? Please take some time and review [Java](https://google.github.io/styleguide/javaguide.html) and [Python](https://google.github.io/styleguide/pyguide.html) style guides.

- [x] My IDE is configured to follow the Google [**code styles**](https://google.github.io/styleguide/).

  No? Unsure? -> [configure your IDE](https://github.com/google/google-java-format).

- [x] I have **added tests** to cover my changes. (If you refactored existing code that was well tested you do not have to add tests)
- [x] I ran `mvn clean package` right before creating this pull request and added all formatting changes to my commit.
- [x] All new and existing **tests passed**.
- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by ndegwamartin** [comment]
@bashir2 a few things to note with the PR.

1. The changes `doc/design.md` documentation are because of applying spotless, let me know if you need me to revert this.
2. Should we care about case sensitivity when performing the check. There may be a performance implication. So something like `acceptEncodingValue.toLowerCase(Locale.ENGLISH).contains(GZIP_ENCODING_VALUE);`

File: doc/design.md [hunk 1]
```diff
@@ -367,7 +367,7 @@ The mapping from resources to patients is done through the
 [patient compartment](https://www.hl7.org/fhir/compartmentdefinition-patient.html)
 definition. Note that we can still access many resources in one query; in
 particular through
-[Patient/ID/$everything](https://hl7.org/fhir/patient-operation-everything.html)
+[Patient/ID/\$everything](https://hl7.org/fhir/patient-operation-everything.html)
```
Comment (bashir2) [inline doc/design.md:370]: nit: is the added `\` required?

File: server/src/main/java/com/google/fhir/gateway/BearerAuthorizationInterceptor.java [hunk 2]
```diff
@@ -211,7 +211,7 @@ private boolean sendGzippedResponse(ServletRequestDetails requestDetails) {
     if (acceptEncodingValue == null) {
       return false;
     }
-    return GZIP_ENCODING_VALUE.equalsIgnoreCase(acceptEncodingValue);
+    return acceptEncodingValue.contains(GZIP_ENCODING_VALUE);
```
Comment (bashir2) [inline server/src/main/java/com/google/fhir/gateway/BearerAuthorizationInterceptor.java:214]: As you suggested, I think it is better to do case-insensitive comparison. Please also update/add unit-tests [here](https://github.com/google/fhir-gateway/blob/4376a58fa2e1316050177c3bb3d206f1f1fcd7c1/server/src/test/java/com/google/fhir/gateway/BearerAuthorizationInterceptorTest.java#L311).

**Review by bashir2 — APPROVED** [review]
Thanks for the fixes @ndegwamartin ; just some minor suggestions.

File: doc/design.md (hunk 1 above)
Comment (bashir2) [reply doc/design.md:370]: BTW, if spotless adds this, please feel free to keep it.

**Pull request merged**